geomstats/geometry/functions.py

```diff
@@ -0,0 +1,245 @@
+"""Module for function spaces as geometric objects"""
+
+import math
+
+import numpy as np
+
+import geomstats.backend as gs
+from geomstats.geometry.manifold import Manifold
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+
+class SinfSpaceMetric(RiemannianMetric):
+    r"""A Riemannian metric on the S_{\inf} space
+
+    Parameters:
+    -------
+    domain_samples : array of shape (n_samples, )
+        grid points on the domain
+    """
+
+    def __init__(self, domain_samples):
+        self.domain = domain_samples
+        self.x = (self.domain - min(self.domain)) / (
+            max(self.domain) - min(self.domain)
+        )
+        self.n_evals = len(self.domain)
+        super().__init__(dim=self.n_evals)
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point=None):
+        """Inner product between two tangent vectors at a base point.
+
+        Parameters
+        ----------
+        tangent_vec_a: array-like, shape=[..., n_evals]
+            Tangent vector at base point.
+        tangent_vec_b: array-like, shape=[..., n_evals]
+            Tangent vector at base point.
+        base_point: array-like, shape=[..., n_evals]
+            Base point.
+            Optional, default: None.
+
+        Returns
+        -------
+        inner_product : array-like, shape=[...,]
+            Inner-product.
+        """
+
+        def _f(v1, v2):
+            return gs.cast(np.trapz(v1 * v2, x=self.x), gs.float32)
```

Comment:
The `trapz` method could be added to the backend, see the equivalent in [torch](https://pytorch.org/docs/stable/generated/torch.trapezoid.html) and [tensorflow](https://www.tensorflow.org/probability/api_docs/python/tfp/math/trapz).